src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -0,0 +1,995 @@
+                return (T)(object)GetSecurityCapabilities(context);
+            }
+            else if (typeof(T) == typeof(IdentityVerifier))
+            {
+                throw new NotImplementedException();
+               // return (T)(object)this.localClientSettings.IdentityVerifier;
+            }
+            else
+            {
+                return context.GetInnerProperty<T>();
+            }
+        }
+
+        internal abstract ISecurityCapabilities GetIndividualISecurityCapabilities();
+
+        ISecurityCapabilities GetSecurityCapabilities(BindingContext context)
+        {
+            ISecurityCapabilities thisSecurityCapability = this.GetIndividualISecurityCapabilities();
+            ISecurityCapabilities lowerSecurityCapability = context.GetInnerProperty<ISecurityCapabilities>();
+            if (lowerSecurityCapability == null)
+            {
+                return thisSecurityCapability;
+            }
+            else
+            {
+                bool supportsClientAuth = thisSecurityCapability.SupportsClientAuthentication;
+                bool supportsClientWindowsIdentity = thisSecurityCapability.SupportsClientWindowsIdentity;
+                bool supportsServerAuth = thisSecurityCapability.SupportsServerAuthentication || lowerSecurityCapability.SupportsServerAuthentication;
+                ProtectionLevel requestProtectionLevel = ProtectionLevelHelper.Max(thisSecurityCapability.SupportedRequestProtectionLevel, lowerSecurityCapability.SupportedRequestProtectionLevel);
+                ProtectionLevel responseProtectionLevel = ProtectionLevelHelper.Max(thisSecurityCapability.SupportedResponseProtectionLevel, lowerSecurityCapability.SupportedResponseProtectionLevel);
+                return new SecurityCapabilities(supportsClientAuth, supportsServerAuth, supportsClientWindowsIdentity, requestProtectionLevel, responseProtectionLevel);
+            }
+        }
+        void SetIssuerBindingContextIfRequired(BindingContext issuerBindingContext)
+        {
+            SetIssuerBindingContextIfRequired(this.EndpointSupportingTokenParameters, issuerBindingContext);
+            SetIssuerBindingContextIfRequired(this.OptionalEndpointSupportingTokenParameters, issuerBindingContext);
+            foreach (SupportingTokenParameters parameters in this.OperationSupportingTokenParameters.Values)
+            {
+                SetIssuerBindingContextIfRequired(parameters, issuerBindingContext);
+            }
+            foreach (SupportingTokenParameters parameters in this.OptionalOperationSupportingTokenParameters.Values)
+            {
+                SetIssuerBindingContextIfRequired(parameters, issuerBindingContext);
+            }
+        }
+       
+
+        protected static void SetIssuerBindingContextIfRequired(SecurityTokenParameters parameters, BindingContext issuerBindingContext)
+        {
+            Type typ = parameters.GetType();
+            if (!(parameters is UserNameSecurityTokenParameters))
```

Comment:
typ isn't used. Also I don't think throwing for the negative case is really necessary.